tests/testthat/_snaps/type.md

```diff
@@ -0,0 +1,151 @@
+# type style
+
+    Code
+      cli_text("{.type {mtcars}}")
```

Comment:
I think writing `{.desc {arg}}` is less intuitive than `{.type {arg}}`. What else would we use `{.type}` for, instead of formatting the type of an object?

Reply:
It could also stand for things like `vctrs::vec_ptype_full()` and `vec_ptype_abbr()`.

Reply:
Those seem much less likely to be useful in cli messages, though. They are also quite unlikely to be in cli, because then cli would need to import vctrs.

Reply:
These are just examples of things that could eventually live in rlang or in a type checking package. Also it would be fine for cli to just throw an error if rlang or another package is not installed. In the future cli should probably call an exported function to get the friendly type (if the implementation is not moved to cli).

Reply:
> Also it would be fine for cli to just throw an error if rlang or another package is not installed.

I don't think that's fine, imagine if it happens while printing an error message. In general `{.type}` should never throw an error. (That's another reason why the rlang implementation is not great for cli, btw.)

Reply:
I think it's fine because it'd be a very rare error and you could display the original error message along with it, using `typeof()` as a fallback.

Reply:
Re the original issue, I'll rename it to `.typeof`, that's mostly what it is, anyway.

Reply:
`typeof` sounds very low level and evokes the base function `typeof()` which returns a type label rather than a type description (i.e. it's more like `vec_ptype_full()`), so it doesn't seem like an improvement.

Reply:
I'll rename it to `.desc` then.

Reply:
Or `.obj_friendly_type` is probably most accurate.

Reply:
It could also be the simpler `.friendly_type`?